### PR TITLE
MONGOID-4919 Use Mongo::Error::SessionsNotSupported for checking session support


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID-4919

File: lib/mongoid/clients/sessions.rb [hunk 1]
```diff
@@ -41,12 +41,20 @@ module Sessions
...
+        end
         session = persistence_context.client.start_session(options)
         Threaded.set_session(session)
         yield(session)
       rescue Mongo::Error::InvalidSession => ex
-        if ex.message == Mongo::Session::SESSIONS_NOT_SUPPORTED
+        if
+          # Driver 2.13.0+
+          defined?(Mongo::Error::SessionsNotSupported) &&
+            Mongo::Error::SessionsNotSupported === ex or
```
Comment (egiurleo) [inline lib/mongoid/clients/sessions.rb:54]: Is there a reason to use `or` here vs `||`? All things being equal, I think I'd prefer consistent styles.

Comment (p-mongo) [reply lib/mongoid/clients/sessions.rb:54]: I think I did it for emphasis but I can redo with punctuation.

Comment (egiurleo) [reply lib/mongoid/clients/sessions.rb]: Thanks!

**Pull request merged**